perceiver_pytorch/queries.py

```diff
@@ -0,0 +1,114 @@
+import torch
+from torch.distributions import uniform
+from typing import List, Union, Tuple
+from perceiver_pytorch.utils import encode_position
+from math import prod
+import einops
+import logging
+
+_LOG = logging.getLogger("perceiver.queries")
+_LOG.setLevel(logging.WARN)
+
+
+class LearnableQuery(torch.nn.Module):
+    """
+    Module that constructs a learnable query of query_shape for the Perceiver
+    """
+
+    def __init__(
+        self,
+        channel_dim: int,
+        query_shape: Union[Tuple[int], List[int]],
+        conv_layer: str = "3d",
+        max_frequency: float = 16.0,
+        num_frequency_bands: int = 64,
+        frequency_base: float = 2.0,
+        sine_only: bool = False,
+    ):
+        """
+        Learnable Query with some inbuilt randomness to help with ensembling
+
+        Args:
+            channel_dim: Channel dimension for the output of the network
+            query_shape: The final shape of the query, generally, the (T, H, W) of the output
+        """
+        super(LearnableQuery, self).__init__()
```

Comment:
What does this do? Sorry - I feel like I should know this

Reply:
I'm not entirely sure, but it is how you are supposed to subclass the `torch.nn.Module` [Doc](https://pytorch.org/docs/stable/generated/torch.nn.Module.html)

Reply:
In Python 3 you can shorten this to:

```suggestion
        super().__init__()
```

I've submitted a [pull request](https://github.com/pytorch/pytorch/pull/65748) to change the PyTorch documentation :)